File: gigachadsys-dao/src/main/java/pe/edu/pucp/gigachadsys/dao/inter/usuarios/EntrenadorDAO.java
package pe.edu.pucp.gigachadsys.dao.inter.usuarios;

import pe.edu.pucp.gigachadsys.dao.base.BaseDAO;
import pe.edu.pucp.gigachadsys.model.personas.Administrador;
import pe.edu.pucp.gigachadsys.model.personas.Entrenador;

import java.util.List;

public interface EntrenadorDAO extends BaseDAO<Entrenador, Integer> {
    List<Entrenador> listAll();
    public Entrenador save(Entrenador entrenador);
    public Entrenador load(Integer id);
    public Entrenador update(Entrenador entrenador);
    public void remove(Entrenador entrenador);
}

File: C#/GigaChadSys/GigaChadSys-DAO/EntrenadorDAO.cs
using GigaChadSysModel;

namespace GigaChadSysDAO;

public interface EntrenadorDAO
{
    List<Entrenador> ListAll();
    Entrenador? Load(int id);
    Entrenador? LoadByEmail(string email);
    Entrenador Save(Entrenador entrenador);
    Entrenador? Update(Entrenador entrenador);
    void Remove(Entrenador entrenador);
}